packages/web/src/ui/elements/UIElement.ts

```diff
@@ -0,0 +1,31 @@
+import { UIElementLocation } from '../locations';
+import { UIElementList } from './UIElementList';
+
+// todo: should this be generic? UIElement<Element<'async'>> ?
+export interface UIElement {
+    nativeElement(): any
+
+    location(): UIElementLocation;
+    locateChildElementAt(locator: UIElementLocation): Promise<UIElement>;
+    locateAllChildElementsAt(locator: UIElementLocation): Promise<UIElementList>;
+
+    enterValue(value: string | number | Array<string | number>): Promise<void>;
+    clearValue(): Promise<void>;
+    click(): Promise<void>;
+    doubleClick(): Promise<void>;
+    moveTo(): Promise<void>;
+    rightClick(): Promise<void>;    // todo: should this be a click() call with a parameter?
```

Comment:
I tend to agree that should be a `click` with a parameter - would be easier to maintain IMO

Reply:
I think we'll need to have a look at Selenium 4, Playwright and WebdriverIO APIs and see what the common denominator is. It feels like `click()` should at the very least accept information about the mouse button to be clicked and maybe (?) the offset to? Not sure if that's useful...